deeplake/api/dataset.py

```diff
@@ -446,34 +523,117 @@ def load(
                 f"A Deep Lake dataset does not exist at the given path ({path}). Check the path provided or in case you want to create a new dataset, use deeplake.empty()."
             )
 
+        if access_method == "stream":
+            dataset_kwargs = {
+                "path": path,
+                "address": address,
+                "storage": cache_chain,
+                "read_only": read_only,
+                "token": token,
+                "org_id": org_id,
+                "verbose": verbose,
+            }
+        else:
+            dataset_kwargs = {
+                "access_method": access_method,
+                "path": path,
+                "read_only": read_only,
+                "memory_cache_size": memory_cache_size,
+                "local_cache_size": local_cache_size,
+                "creds": creds,
+                "token": token,
+                "org_id": org_id,
+                "verbose": verbose,
+                "ds_exists": True,
+                "num_workers": num_workers,
+                "scheduler": scheduler,
+                "reset": reset,
+            }
+
         try:
+            return dataset._load(dataset_kwargs, access_method)
+        except (AgreementError, CheckoutError, LockedException) as e:
+            raise e from None
+        except Exception as e:
             if access_method == "stream":
-                ret = dataset_factory(
-                    path=path,
-                    storage=cache_chain,
-                    read_only=read_only,
-                    token=token,
-                    org_id=org_id,
-                    verbose=verbose,
+                if not reset:
+                    if isinstance(e, DatasetCorruptError):
+                        raise DatasetCorruptError(
+                            message=e.message,
+                            action="Try using `reset=True` to reset HEAD changes and load the previous commit.",
+                            cause=e.__cause__,
+                        )
+                    raise DatasetCorruptError(
+                        "Exception occured (see Traceback). The dataset maybe corrupted."
+                        "Try using `reset=True` to reset HEAD changes and load the previous commit."
+                        "This will delete all uncommitted changes on the branch you are trying to load."
+                    ) from e
+                if storage.read_only:
+                    raise ReadOnlyModeError(
+                        "Cannot reset when loading dataset in read-only mode."
```

Comment:
in this case then the user should be able to go back to previous commit. We can print previous commit_id and ask user to specify that